$OpenBSD: patch-gnucash_gnome-utils_gnc-component-manager_c,v 1.1 2020/08/24 08:01:49 ajacoutot Exp $

https://bugs.gnucash.org/show_bug.cgi?id=797921
error: result of comparison against a string literal is unspecified (use an explicit string comparison function instead) [-Werror,-Wstring-compare]

Index: gnucash/gnome-utils/gnc-component-manager.c
--- gnucash/gnome-utils/gnc-component-manager.c.orig
+++ gnucash/gnome-utils/gnc-component-manager.c
@@ -267,7 +267,14 @@ gnc_cm_event_handler (QofInstance *entity,
 #endif
     add_event (&changes, guid, event_type, TRUE);
 
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wstring-compare"
+#endif
     if (QOF_CHECK_TYPE(entity, GNC_ID_SPLIT))
+#ifdef __clang__
+#pragma clang diagnostic pop
+#endif
     {
         /* split events are never generated by the engine, but might
          * be generated by a backend (viz. the postgres backend.)
